internal/controller/policyserver_controller_deployment.go

```diff
@@ -345,6 +252,181 @@ func (r *PolicyServerReconciler) adaptDeploymentSettingsForPolicyServer(policySe
 	}
 }
 
+func configureImagePullSecret(policyServer *policiesv1.PolicyServer, admissionContainer *corev1.Container) {
+	if policyServer.Spec.ImagePullSecret != "" {
+		admissionContainer.VolumeMounts = append(admissionContainer.VolumeMounts,
+			corev1.VolumeMount{
+				Name:      imagePullSecretVolumeName,
+				ReadOnly:  true,
+				MountPath: dockerConfigJSONPolicyServerPath,
+			})
+		admissionContainer.Env = append(admissionContainer.Env,
+			corev1.EnvVar{
+				Name:  "KUBEWARDEN_DOCKER_CONFIG_JSON_PATH",
+				Value: dockerConfigJSONPolicyServerPath,
+			})
+	}
+}
+
+func configuresInsecureSources(policyServer *policiesv1.PolicyServer, admissionContainer *corev1.Container) {
+	if len(policyServer.Spec.InsecureSources) > 0 || len(policyServer.Spec.SourceAuthorities) > 0 {
+		admissionContainer.VolumeMounts = append(admissionContainer.VolumeMounts,
+			corev1.VolumeMount{
+				Name:      sourcesVolumeName,
+				ReadOnly:  true,
+				MountPath: constants.PolicyServerSourcesConfigContainerPath,
+			})
+		admissionContainer.Env = append(admissionContainer.Env,
+			corev1.EnvVar{
+				Name:  "KUBEWARDEN_SOURCES_PATH",
+				Value: filepath.Join(constants.PolicyServerSourcesConfigContainerPath, sourcesFilename),
+			})
+	}
+}
+
+func configureLabelsAndAnnotations(policyServerDeployment *appsv1.Deployment, policyServer *policiesv1.PolicyServer, configMapVersion string) {
+	if policyServerDeployment.ObjectMeta.Annotations == nil {
+		policyServerDeployment.ObjectMeta.Annotations = make(map[string]string)
+	}
+	policyServerDeployment.ObjectMeta.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation] = configMapVersion
+
+	if policyServerDeployment.Labels == nil {
+		policyServerDeployment.Labels = make(map[string]string)
+	}
+	policyServerDeployment.Labels[constants.AppLabelKey] = policyServer.AppLabel()
+	policyServerDeployment.Labels[constants.PolicyServerLabelKey] = policyServer.Name
+}
+
+func (r *PolicyServerReconciler) configureMutualTLS(ctx context.Context, policyServerDeployment *appsv1.Deployment) error {
+	policyServerDeployment.Spec.Template.Spec.Volumes = append(
+		policyServerDeployment.Spec.Template.Spec.Volumes,
+		corev1.Volume{
+			Name: kubewardenCAVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: constants.CARootSecretName,
+					Items: []corev1.KeyToPath{
+						{
+							Key:  constants.CARootCert,
+							Path: constants.CARootCert,
+						},
+					},
+				},
+			},
+		},
+	)
+
+	admissionContainer := &policyServerDeployment.Spec.Template.Spec.Containers[0]
+	if r.ClientCAConfigMapName != "" {
+		if err := r.Client.Get(ctx, types.NamespacedName{Name: r.ClientCAConfigMapName, Namespace: r.DeploymentsNamespace}, &corev1.ConfigMap{}); err != nil {
+			return fmt.Errorf("failed to fetch client CA config map: %w", err)
+		}
```

Comment:
I think it's nice for us to bubble up this error. We could also have ignored that, the container would have stayed in status "pending" forever because the ConfigMap could not be mounted

Reply:
Yes, it is a better UX to bubble up the error. We do the same for the CA/certs secret in the other suberconciler.